qcfractal/interface/client.py

```diff
@@ -200,64 +254,60 @@ def server_information(self) -> Dict[str, str]:
 
 ### KVStore section
 
-    def query_kvstore(self, id: List[str], full_return: bool=False) -> Dict[str, Any]:
+    def query_kvstore(self, id: QueryObjectId, full_return: bool=False) -> Dict[str, Any]:
         """Queries items from the database's KVStore
 
         Parameters
         ----------
-        id : List[str]
+        id : QueryObjectId
             A list of KVStore id's
         full_return : bool, optional
-            Optionally returns the full request result including the ``meta`` field.
+            Returns the full server response if True that contains additional metadata.
 
         Returns
         -------
         Dict[str, Any]
             A list of found KVStore objects in {"id": "value"} format
         """
-        body = KVStoreGETBody(data=id, meta={})
-        r = self._request("get", "kvstore", data=body.json())
-        r = KVStoreGETResponse.parse_raw(r.text)
 
-        if full_return:
-            return r
-        else:
-            return r.data
+        return self._automodel_request("kvstore", "get", {"meta": {}, "data": id}, full_return=full_return)
 
 ### Molecule section
 
     def query_molecules(self,
-                        id: Optional[List[str]]=None,
-                        molecule_hash: Optional[List[str]]=None,
-                        molecular_formula: Optional[List[str]]=None,
+                        id: QueryObjectId=None,
+                        molecule_hash: QueryStr=None,
+                        molecular_formula: QueryStr=None,
                         full_return: bool=False) -> List[Molecule]:
         """Queries molecules from the database.
 
         Parameters
         ----------
-        id : Optional[List[str]], optional
+        id : QueryObjectId, optional
             Queries the Molecule ``id`` field.
-        molecule_hash : Optional[List[str]], optional
+        molecule_hash : QueryStr, optional
             Queries the Molecule ``molecule_hash`` field.
-        molecular_formula : Optional[List[str]], optional
+        molecular_formula : QueryStr, optional
             Queries the Molecule ``molecular_formula`` field.
         full_return : bool, optional
-            Optionally returns the full request result including the ``meta`` field.
+            Returns the full server response if True that contains additional metadata.
 
         Returns
         -------
-        Dict[str, 'Molecule']
+        List[Molecule]
```

Comment:
General thought, do we want to make the returns a Union between `List[Molecule]` and `Dict[str, Any]` because of the `full_return` option? We don't even before this PR, but its a thought. Maybe change the documentation?

Reply:
I would like to just give return types for the non full option. Complex otherwise.